src/gauge/radial-gauge/radial-gauge.component.ts

```diff
@@ -0,0 +1,266 @@
+  @Input() public value: number = 20;
+  @Input() public minValue = 0;
+  @Input() public maxValue = 100;
+
+  @Input() public minAngle = -90;
+  @Input() public maxAngle = 90;
+  @Input() public innerArcRadius: number;
+  @Input() public outerArcRadius: number;
+
+  @Input() public majorTicks = 5;
+  @Input() public axisRadius: number;
+
+  @Input() public pointerWidth: number;
+  @Input() public pointerHeadLength: number;
+  @Input() public pointerTailLength: number;
+  @Input() public pointerColor: string;
+  @Input() public segments = [];
+
+  @Output() public click: EventEmitter<any> = new EventEmitter();
+  @Output() public mouseEnter: EventEmitter<any> = new EventEmitter();
+  @Output() public mouseLeave: EventEmitter<any> = new EventEmitter();
+
+  @Output() public arcClick: EventEmitter<any> = new EventEmitter();
+  @Output() public arcMouseEnter: EventEmitter<any> = new EventEmitter();
+  @Output() public arcMouseLeave: EventEmitter<any> = new EventEmitter();
+
+  @Output() public labelClick: EventEmitter<any> = new EventEmitter();
+  @Output() public labelMouseEnter: EventEmitter<any> = new EventEmitter();
+  @Output() public labelMouseLeave: EventEmitter<any> = new EventEmitter();
+
+  @Output() public pointerClick: EventEmitter<any> = new EventEmitter();
+  @Output() public pointerMouseEnter: EventEmitter<any> = new EventEmitter();
+  @Output() public pointerMouseLeave: EventEmitter<any> = new EventEmitter();
+
+  private colors: ColorHelper;
+  private ticks: number[];
+  private degreeRange: number;
+
+  ngAfterViewInit(): void {
+    super.ngAfterViewInit();
+  }
+
+  public update(): void {
+    super.update();
+
+    this.ticks = this.getTicks();
+    this.degreeRange = this.getDegreeRange();
+
+    this.dimensions = this.getDimensions();
+
+    this.displayingValue = !this.checkStringEmpty(this.displayValue) 
+    ? this.displayValue 
+    : this.value.toString();
+
+    this.colors = new ColorHelper(this.scheme, this.schemeType, 
+      [this.minValue, this.maxValue], this.customColors);
+
+    this.arcs = this.getArcs();
+  }
+
+  public getTranslate(): string {
+    const twidth = this.width / 2;
+    const theight = this.height / 2;
+    return `translate(${twidth}, ${theight})`;
+  }
+
+  public movePointer(): string {
+    return `rotate(${this.getPointerLocation()})`;
+  }
+
+  public getTextTransform(): string {
+    const scale: number = this.getElementScale(0.004);
+    return `scale(${scale}, ${scale})`;
+  }
+
+  public getPointer(): string {
+    const pointerLine: Line<[number, number]> = line().curve(curveLinear);
+    return pointerLine(this.getPointerData());
+  }
+
+  public getScale(): ScaleLinear<number, number> {
+    return scaleLinear()
+    .range([0, this.getDegreeRange()])
+    .nice()
+    .domain([this.minValue, this.maxValue]);
+  }
+
+  public getArcs(): any {
+    const result = [];
+
+    if (this.segments != null && this.segments.length > 0) {
+      for(const i in this.segments) {
```

Comment:
I might be missing something simple, but searching through the code for usage/mutations of `this.segments`, I'm not seeing any, and adding a break-point here I did not see it get hit inside this initial `if`. Other than it being initialized to an empty array. I think this branching logic is dead code that can all be safety deleted within this if.